numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2459,46 +2458,48 @@ einsum_list_to_subscripts(PyObject *obj, char *subscripts, int subsize)
             subscripts[subindex++] = '.';
             ellipsis = 1;
         }
-        /* Subscript */
-        else if (PyInt_Check(item) || PyLong_Check(item)) {
-            long s = PyInt_AsLong(item);
-            npy_bool bad_input = 0;
+        else {
+            long s = PyLong_AsLong(item);
```

Comment:
What error message does this give if you pass a float? If it doesn't error at all, you might need to change to the following:
```suggestion
            Py_ssize s = PyNumber_Index(item);
```

Reply:
You're right it didn't give an error on floats. I think I have fixed that now. Hopefully 

`item = PyNumber_Index(item);` 

is OK? Since it just modifies the object. I'm very new to the C Python API.

Reply:
Oh, I guess I forgot the return type. I think we might have a helper function that does what we need here, perhaps `PyArray_IntAsIntp` or something similar.

Reply:
Yes, which is much the same as `PyNumber_AsSsize_t ` but rejects bools.

Reply:
I have edited the function to use `PyArray_PyIntAsIntp` instead, which is what I believe you were referencing. It did used to accept bools before, but now it doesn't.

Reply:
Yeah, I am not quite sure. Doesn't feel quite right to switch over silently, but on the other hand it seems not like a much used corner and then a corner case of that.

Can you add those new tests in either case? They are necessary whether or not we let bools pass.